scripts/algcomptool/AlgCompTool.py

```diff
@@ -0,0 +1,335 @@
+
+		#Comparison of algorithms for when both predicted correctly
+		fileComp.write("\nCOMPARISON OF ALGS WHEN BOTH PREDICTED CORRECTLY\n")
+		for subject in bothCorrect:
+
+			alg1CompSegment = alg1[alg1['sub'] == subject]
+			alg2CompSegment = alg2[alg2['sub'] == subject]
+
+			#Same prediction level, so must compare ratios between our first correct prediction, and the next prediction
+			alg1Ratio = alg1CompSegment.iloc[0,scoreColLocation]/alg1CompSegment.iloc[1,scoreColLocation]
+			alg2Ratio = alg2CompSegment.iloc[0,scoreColLocation]/alg2CompSegment.iloc[1,scoreColLocation]
+
+			#The greater the ratio the better the prediction, i.e. 
+			#0.5 (our prediction) 0.04 (next prediction) vs.
+			#0.5 (our prediction) 0.4 (next prediction)
+			if alg1Ratio > alg2Ratio:
+				fileComp.write("Alg1 > Alg2 for relation: " + " ID: " + str(idSubjectMapping[subject]) + " - Subject: " + str(subject.replace("dbr:","")) + "\n")
+			elif alg1Ratio < alg2Ratio:
+				fileComp.write("Alg1 < Alg2 for relation: " + " ID: " + str(idSubjectMapping[subject]) + " - Subject: " + str(subject.replace("dbr:","")) + "\n")
+
+		fileComp.write("\nCOMPARISON OF ALGS WHEN BOTH PREDICTED INCORRECTLY\n")
+		#Comparison of algorithms for when both predicted incorrectly
+		for subject in bothIncorrect:
+
+			#Alg1 Analysis
+			alg1CompSegment = alg1[alg1['sub'] == subject]
+			alg1Sample = 0
+			while alg1CompSegment.iloc[alg1Sample,classColLocation] != 1:
+				alg1Sample = alg1Sample + 1
+
+			#Alg2 Analysis
+			alg2CompSegment = alg2[alg2['sub'] == subject]
+			alg2Sample = 0
+			while alg2CompSegment.iloc[alg2Sample,classColLocation] != 1:
+				alg2Sample = alg2Sample + 1
+
+			if alg1Sample > alg2Sample:
+				fileComp.write("Alg1 < Alg2 for relation: " + " ID: " + str(idSubjectMapping[subject]) + " - Subject: " + str(subject.replace("dbr:","")) + "\n")
+			elif alg1Sample < alg2Sample:
+				fileComp.write("Alg1 > Alg2 for relation: " + " ID: " + str(idSubjectMapping[subject]) + " - Subject: " + str(subject.replace("dbr:","")) + "\n")
+			elif alg1Sample == alg2Sample:
+				#Same prediction level, so must compare ratios between highest weight prediction and this wrong preiction
+				alg1Ratio = alg1CompSegment.iloc[0,scoreColLocation]/alg1CompSegment.iloc[alg1Sample,scoreColLocation]
+				alg2Ratio = alg2CompSegment.iloc[0,scoreColLocation]/alg2CompSegment.iloc[alg2Sample,scoreColLocation]
+				#The bigger the ratio the worse the prediction: i.e. 0.5 (first), 0.3, 0.001, 0.0001 (our prediction) vs.
+				#0.5 (first), 0.2 (our prediction)
+				if alg1Ratio > alg2Ratio:
+					fileComp.write("Alg1 < Alg2 for relation: " + " ID: " + str(idSubjectMapping[subject]) + " - Subject: " + str(subject.replace("dbr:","")) + "\n")
+				elif alg1Ratio < alg2Ratio:
+					fileComp.write("Alg1 > Alg2 for relation: " + " ID: " + str(idSubjectMapping[subject]) + " - Subject: " + str(subject.replace("dbr:","")) + "\n")
+		fileComp.close
```

Comment:
Changing all "missing function calls" you mentioned except this one, because this requires a different output format.